central/reports/scheduler/schedule.go

```diff
@@ -364,24 +385,42 @@ func formatMessage(rc *storage.ReportConfiguration, emailTemplate string, date t
 }
 
 func (s *scheduler) getReportData(ctx context.Context, rQuery *common.ReportQuery) ([]common.Result, error) {
-	r := make([]common.Result, 0, len(rQuery.ScopeQueries))
-	for _, sq := range rQuery.ScopeQueries {
-		resultData, err := s.runPaginatedQuery(ctx, sq, rQuery.CveFieldsQuery)
-		if err != nil {
-			return nil, err
+	if !features.ObjectCollections.Enabled() {
+		r := make([]common.Result, 0, len(rQuery.ScopeQueries))
+		for _, sq := range rQuery.ScopeQueries {
+			resultData, err := s.runPaginatedQuery(ctx, sq, rQuery.CveFieldsQuery, rQuery.DeploymentsQuery)
+			if err != nil {
+				return nil, err
+			}
+			r = append(r, resultData)
 		}
-		r = append(r, resultData)
+		return r, nil
 	}
-	return r, nil
+	result, err := s.runPaginatedQuery(ctx, "", rQuery.CveFieldsQuery, rQuery.DeploymentsQuery)
+	if err != nil {
+		return nil, err
+	}
+	result = groupByClusterAndNamespace(result)
+	return []common.Result{result}, nil
 }
 
-func (s *scheduler) runPaginatedQuery(ctx context.Context, scopeQuery, cveQuery string) (common.Result, error) {
+func (s *scheduler) runPaginatedQuery(ctx context.Context, scopeQuery, cveQuery string, deploymentsQuery *v1.Query) (common.Result, error) {
 	offset := 0
 	var resultData common.Result
 	for {
 		var gqlQuery string
+		gqlPaginationOffset := offset
 		if env.PostgresDatastoreEnabled.BooleanSetting() {
 			gqlQuery = reportQueryPostgres
+			if features.ObjectCollections.Enabled() {
+				deploymentIds, err := s.getDeploymentIDs(ctx, deploymentsQuery, int32(offset))
+				if err != nil || len(deploymentIds) == 0 {
+					return common.Result{}, err
+				}
+				scopeQuery = fmt.Sprintf("%s:%s", search.DeploymentID.String(), strings.Join(deploymentIds, ","))
+				log.Infof("ROX-12629 : deployments scopeQuery : %s", scopeQuery)
```

Comment:
remove? debug?

Reply:
Yeah, had added it for debugging to see if query isn't too long, will remove it.